libs/dao/src/main/java/com/akto/dto/ApiCollectionUsers.java

```diff
@@ -56,6 +68,108 @@ public enum CollectionType {
             });
         }});
 
+    public static Set<String> getUrlsFromConditions(List<CollectionCondition> conditions){
+        Set<ApiInfoKey> apis = new HashSet<>();
+        conditions.forEach((condition) -> {
+            if(CollectionCondition.Operator.AND.equals(condition.getOperator())){
+                apis.retainAll(condition.returnApis());
```

Comment:
`apis` defaults to empty set. `retainAll` will be a no-op here

Reply:
You are correct. This is fixed in the followup PR. https://github.com/akto-api-security/akto/pull/740/files#diff-e8f80dd82161897bd4f7ea61b0d34f3ad0c3cf73cf8568ecebb054ef37fb3a85R84